app/src/command.rs

```diff
@@ -48,15 +58,27 @@ impl<'app, D: Dependencies<'app>> Application<'app, D> {
             .iter()
             .cloned()
             .chain(self.mem_pool.drain().map(Into::into))
-            .filter(|tx|
+            .filter_map(|tx|
                 // Do not include transactions we have already processed before
-                !self.receipt_repository.contains(&self.receipt_memory, tx.tx_hash()).unwrap())
-            .collect::<Vec<_>>();
+                match self.receipt_repository.contains(&self.receipt_memory, tx.tx_hash()) {
+                    Ok(false) => Some(Ok(tx)),
+                    Ok(true) => None,
+                    Err(e) => Some(Err(e)),
+                }
+            )
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| {
+                tracing::error!("Failure during `start_block_build`. Receipt queries failed: {e:?}");
+                UnrecoverableAppFailure
+            })?;
         let parent = self
             .block_repository
             .latest(&self.storage)
-            .unwrap()
-            .expect("Parent block should exist");
+            .map_err(|e| {
+                tracing::error!("Failure during `start_block_build`. Failed to get latest block from block repository: {e:?}");
+                UnrecoverableAppFailure
+            })?
+            .expect("Block repository is non-empty (must always at least contain genesis)");
```

Comment:
Is the first transaction always the L1 deposit transaction instead of genesis? By genesis did you mean that?